cli/plugins/docker/index.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+class Docker {
+  constructor(joy) {
+    const prog = joy.prog;
+    prog
+      .command('stack', 'Build, start, stop, restart the Docker stack')
+      .argument('<subcommand>', 'subcommand', ['build', 'start', 'stop', 'restart'])
+      .option('-n, --name <name>', 'Component name', prog.STRING)
+      .action(async (args, options, logger) => {
+        logger.debug("arguments: %j", args);
+        logger.debug("options: %j", options);
+        const code = await this[args.subcommand].call(joy, options, logger);
+        return code;
+      });
+  }
+
+
+  /**
+   * Returns a Promise from invoking an external command to `docker-compose up build`
+   * 
+   */
+  start() {
+    return this.invoke('docker-compose', ['-f', `.joy/docker/docker-compose.yml`, 'up', '-d']);
+  }
+
+
+  /**
+   * Returns a Promise from invoking an external command to `docker-compose down`
+   * 
+   */
+  stop() {
+    return this.invoke('docker-compose', ['-f', `.joy/docker/docker-compose.yml`, 'down']);
+  }
+
+
+  /**
+   * Returns a Promise from invoking an external command to `docker build`
+   * 
+   * @param {*} options 
+   * @param {*} logger 
+   */
+  build(options, logger) {
+    const imageName = `${process.env.ORG}/${options.name}.${process.env.PRODUCT}.${process.env.TLD}`;
+    return this.invoke('docker', ['build', '--force-rm', '--no-cache', '-f', `.joy/docker/${options.name}.dockerfile`, '.', '-t', imageName])
```

Comment:
Should we enforce semicolons for every line that supports them or only where required?

Reply:
Enforce for all. There's a minor overhead by the JS engine to parse and add semicolons which we can mitigate. This should be an eslint rule.